corehq/apps/ota/utils.py

```diff
@@ -118,6 +118,12 @@ def is_permitted_to_restore(domain, couch_user, as_user, has_data_cleanup_privel
         else:
             if user_domain != domain:
                 message = u"{} was not in the domain {}".format(username, domain)
+
+                # In this case we may be dealing with a WebUser
+                user = WebUser.get_by_username(as_user)
+                if user and domain in user.domains:
+                    message = None
```

Comment:
nit but seems like it'd be more readable if the `message` assignment above went in the `else`